src/kmer_unordered_set.cpp

```diff
@@ -68,50 +96,75 @@ void KmerSets::get_parent_kmer_sets(){
     path relative_hap2_kmer_list_path = "data/hg04.all.homo.unique.kmer.1000.fa";
     path absolute_hap2_kmer_list_path = project_directory / relative_hap2_kmer_list_path;
 
-    load_file_into_unordered_set(absolute_hap1_kmer_list_path,hap1_kmer_set);
-    load_file_into_unordered_set(absolute_hap2_kmer_list_path,hap2_kmer_set);
+    load_file_into_unordered_set(absolute_hap1_kmer_list_path,hap1_kmer_set,"hap1");
+    load_file_into_unordered_set(absolute_hap2_kmer_list_path,hap2_kmer_set,"hap2");
+
+}
+
+
+void KmerSets::print_component_parent_conf_matrix() {
+	cout << "Number of graph components: " << component_map.size() << endl;
+	
+	for (int i = 0; i < component_map.size(); i++) {
+
+		cout << "component: " << i << "\n   pat\tmat" << endl;
+		
+		for (int j = 0; j < 2; j++) {
+			cout << j << " |" << component_map[i][j][parent_hap1_int] << "\t|" << component_map[i][j][parent_hap2_int] << "\t|" <<endl;
+		}
+		cout << endl;
+	}
+}
 
+void KmerSets::parse_path_string(string path_string){
+	int string_delim = path_string.find('-');
+	this->graph_component = stoi(path_string.substr(0,string_delim));
+	this->component_haplotype = stoi(path_string.substr(string_delim+1,path_string.length()));
 }
 
 // single kmer find
-bool KmerSets::find_haplotype_single_kmer_count(string child_kmer) {
-	hap1_kmer_count = 0;
-	hap2_kmer_count = 0;
+bool KmerSets::increment_parental_kmer_count( string path_hap_string, string child_kmer) {
+
+	// this is done for each kmer - consider moving it out of this function
+	parse_path_string(path_hap_string);
```

Comment:
in practice this might actually be plenty fast for our purposes, so lets not worry about it right now. we could always switch back to computing before the loop starts